.pipelines/templates/template-podman.yml

```diff
@@ -0,0 +1,163 @@
+
+          # Check if login was successful
+          if [ $? -ne 0 ]; then
+            echo "Error: Podman login failed."
+            cat podman_login.log
+            kill_podman
+            exit 1
+          fi
+
+          echo "Podman login successful."
+
+          # Run the build and generate a unique image tag
+          echo "Running the build and capturing the image tag..."
+          IMAGE_TAG=$(eval "${{ parameters.makeCommand }}" | grep 'Successfully tagged' | awk '{print $3}' | cut -d':' -f2)
+
+          # If IMAGE_TAG is empty, exit without proceeding to tag and push
+          if [ -z "$IMAGE_TAG" ]; then
+            echo "Warning: No image was built or tagged by the makeCommand. Skipping image push."
+            kill_podman
+            exit 0  # Exit successfully as there's no image to tag or push
+          fi
+
+          # Continue only if an image was successfully built
+          UNIQUE_TAG="${IMAGE_TAG}-$(date +%Y%m%d%H%M%S)-$(Build.BuildId)"
+          podman $PODMAN_REMOTE_ARGS images
+
+          echo "Captured UNIQUE_TAG: $UNIQUE_TAG"
+
+          # Retag the image with the ACR repository and unique tag
+          podman $PODMAN_REMOTE_ARGS tag localhost/${IMAGE_REPO_NAME}:${IMAGE_TAG} ${ACR_FQDN}/${REPOSITORY}:${UNIQUE_TAG}
+
+          # Push the image to ACR
+          podman $PODMAN_REMOTE_ARGS push ${ACR_FQDN}/${REPOSITORY}:${UNIQUE_TAG}
+
+          # Optionally tag and push the image as 'latest'
+          if [ ${{ parameters.pushLatest }} == true ]; then
+            podman $PODMAN_REMOTE_ARGS tag ${ACR_FQDN}/${REPOSITORY}:${UNIQUE_TAG} ${ACR_FQDN}/${REPOSITORY}:latest
+            podman $PODMAN_REMOTE_ARGS push ${ACR_FQDN}/${REPOSITORY}:latest
+          fi
+
+          # Check for images with label 'stage=rp-build-cache-layer' to extract cover.out
+          RP_IMAGE_ID=$(podman $PODMAN_REMOTE_ARGS image ls --filter label=stage=rp-build-cache-layer --noheading --format '{{.Id}}' | tail -n 1)
+
+          if [ -n "$RP_IMAGE_ID" ]; then
+              # If the image exists, extract cover.out
+              echo "Found RP image: $RP_IMAGE_ID"
+              podman $PODMAN_REMOTE_ARGS create --name extract_cover_out $RP_IMAGE_ID
+              podman $PODMAN_REMOTE_ARGS cp extract_cover_out:/app/cover.out $(Build.SourcesDirectory)/cover.out
+              podman $PODMAN_REMOTE_ARGS rm extract_cover_out
+
+              # Ensure cover.out was copied
+              if [ ! -f $(Build.SourcesDirectory)/cover.out ]; then
+                  echo "Error: cover.out file was not extracted successfully."
+                  exit 1
+              else
+                  echo "cover.out file found in $(Build.SourcesDirectory)"
+              fi
+          else
+              # If no image is found, skip the cover.out extraction
+              echo "No RP Image found with label 'stage=rp-build-cache-layer'. Skipping cover.out extraction."
+          fi
```

Comment:
I think we can improve this. The fact we have to install `go` in this stuff here in the pipeline is telling me that `ci-rp` is not doing the same thing as the old build, where we had `make unit-test-go` -> `make xmlcov` -> publish results and coverage files:

```
        - script: |
            set -xe
            make unit-test-go
          displayName: 🧪 Run Golang unit tests
          target: golang

        - task: PublishTestResults@2
          displayName: 📊 Publish tests results
          inputs:
            testResultsFiles: $(System.DefaultWorkingDirectory)/**/report.xml
          condition: succeededOrFailed()
          // **BUG: this needs to run in `target:golang` - how is this even working? (or is it?)

        - script: |
            make xmlcov
          displayName: ⚙️ Process Reports
          condition: succeededOrFailed()
          target: golang

        - task: PublishCodeCoverageResults@1
          displayName: 📈 Publish code coverage
          inputs:
            codeCoverageTool: Cobertura
            summaryFileLocation: $(System.DefaultWorkingDirectory)/**/coverage.xml
            failIfCoverageEmpty: false
          condition: succeededOrFailed()
          target: golang
```

What I'd like to see in the new pipeline is effectively a one line code change (in actuality, we know that it's not "just" one line because of podman setup/etc ... but `make ci-rp` should give us exactly the same local files we expected before. We should be able to streamline the process into: `make ci-rp` -> publish coverage files.

```
        - script: |
            set -xe
            make ci-rp
          displayName: 🧪 Run Golang unit tests
          target: golang

          // rest is the same
```

Can we update the Dockerfile and "extract" steps in the Makefile to get us all the same local files we need, rather than rolling a new process here in the pipe?

Reply:
Please see the latest code with fix